keystore/keystore.go

```diff
@@ -149,5 +153,19 @@ func (ks *FSKeystore) List() ([]string, error) {
 		return nil, err
 	}
 
-	return dir.Readdirnames(0)
+	dirs, err := dir.Readdirnames(0)
+	if err != nil {
+		return nil, err
+	}
+
+	var list []string
+
+	for _, name := range dirs {
+		err := validateName(name)
+		if err == nil {
```

Comment:
Should probably log the error. You'll need to import the logger and construct it at the top of the file:

```go
import (
  // ...
  logging "gx/ipfs/QmRb5jh8z2E8hMGN2tkvs1yHynUanqnZ3UeKwgN1i9P1F8/go-log"
)

var log = logging.Logger("keystore")

// ...
````